### PR TITLE
website: News site

### PR DESCRIPTION
**Pull request by jemgillam**
## Description

 - [x] New news section (a Docusaurus blog)
 - [x] Article about the new release
 - [x] Sponsors page
 - [x] Tidy up release notes
 - [x] Add docs link to navbar
 
Article as workshopped. 

I'm not entirely sure about using the orange on the light version of the news site. It gets AA rating on the contrast checker, it just seems a little jarring to me. Perhaps we can use what I think of as the primary green.  Would be a pain to do though.

I've tidied up the release notes by putting in headings and having it follow the same order as the news article. I've also added a link at the top to the news article as I think it's easier to read as a "digest" and with some of the changes explained more thoroughly. Please check that the lines I've removed are reflected in the lines I've added - I have been conservative with removing lines from the release notes. I've had to put a "misc" section at the end, feel free to move any of them to the prose if they fit. 

Sponsors page shows the featured sponsors and worker project sponsors, the orange/green words under the logo links to the Graphile site. This page should be nicely responsive, though on  mobile the logos take up a lot of scrolling room. 

Sponsors page should be easily updateable.

## Performance impact
NA

## Security impact
NA

## Checklist

- [x] My code matches the project's code style and `yarn lint:fix` passes.
- ~[ ] I've added tests for the new feature, and `yarn test` passes.~
- ~[ ] I have detailed the new feature in the relevant documentation.~
- ~[ ] I have added this feature to 'Pending' in the `RELEASE_NOTES.md` file (if one exists).~
- ~[ ] If this is a breaking change I've explained why.~

### CODE REVIEW

**Comment by jemgillam** [comment]
I don't know if we should change the orange to a (more readable) green? 

Screenshot 2023-11-27 at 12 17 56

**Comment by jemgillam** [comment]
Something weird is going on with the ant-heart svg. The website shows that it has a faint orange line near the ant, I think I can see it in the preview here too. But my inkscape doesn't show it at all 🤷

**Comment by jemgillam** [comment]
Sponsors page

Screenshot 2023-11-30 at 14 47 33

Screenshot 2023-11-30 at 14 48 17

File: website/src/components/SponsorButtons/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          styles.sponsorbutton,
+        )}
+        to="https://github.com/users/benjie/sponsorship"
+      ></Link>
+      <Link
+        className={clsx(
+          "button button--outline button--lg",
+          styles.borderbutton,
+        )}
+        to="https:/graphile.org"
```
Comment (benjie) [inline website/src/components/SponsorButtons/index.js:21]:
```suggestion
        to="https://graphile.org"
```

**Review by benjie — APPROVED** [review]
Awesome!

**Pull request merged**